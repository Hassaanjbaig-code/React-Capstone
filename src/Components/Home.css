.Fetch-home {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  padding: 0;
}

.Fetch-home li {
  max-height: 250px;
  max-width: 300px;
  min-height: 150px;
  min-width: 120px;
  list-style: none;
  padding: 1rem;
  border: 1px solid #d0d0d082;
  width: 37%;
  line-height: 1rem;
}

.Fetch-home li:nth-child(even) {
  background-color: #d15b9d;
}

.Fetch-footer {
  display: flex;
  width: 90%;
  justify-content: space-between;
  position: relative;
  bottom: -14px;
}

.bold {
  font-weight: bold;
  color: #fff;
}

.align {
  text-align: end;
}

.Button {
  border: none;
  background: transparent;
  position: relative;
  left: 98px;
  bottom: -46px;
  font-size: 1.4rem;
  cursor: pointer;
}

.align-item {
  bottom: 18px;
}

.Fetch-footer p {
  display: flex;
  flex-direction: column;
}

/* @media screen and (min-width: 729px) { */
  
/* } */
