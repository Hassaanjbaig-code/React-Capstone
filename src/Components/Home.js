import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { stockdata } from '../redux/Stock-data/stock';
import { NavLink } from 'react-router-dom';
import { IC } from '../redux/Stock-data/IncomeStatement';
import './Home.css';

const Home = () => {
    const dispatch = useDispatch();
    useEffect(() => {
        dispatch(stockdata());
    }, [dispatch])
    const show = useSelector(state => state.stock);
    let count = 0;
  return (
    <div>
        <ul className='Fetch-home'>
        {show.data.map((data) => { 
            count = count + 1;
        return (
            <li id={data.volumn} key={count}>
                <h2>{data.symbol}</h2>
                <h3>{data.companyName}</h3>
                <div className="Fetch-footer">
                <div>
                    <p><span className='bold'>Price:</span> {data.price}</p>
                </div>
                <div className="align">
                    <p> <span className='bold'>Industry:</span> {data.industry}</p>
                </div>
                </div>
                <NavLink to='/IncomeStatement' >
                   <button type="button" onClick={() => dispatch(IC(data.symbol))}>Income Statement</button>
                </NavLink>
            </li>
        )})}
        </ul>
    </div>
  )
}

export default Home
