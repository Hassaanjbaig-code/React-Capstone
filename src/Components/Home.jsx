/* eslint-disable jsx-a11y/control-has-associated-label */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { FaAngleRight } from 'react-icons/fa';
import { FiSearch } from 'react-icons/fi';
import { stockdata } from '../redux/Stock-data/stock';
import { IC } from '../redux/Stock-data/IncomeStatement';
import './Home.css';

const handlefilter = (query, data) => {
  if (query.length === 0) {
    return data;
  }
  return data
    .filter((data) => data.companyName.toLowerCase().includes(query.toLowerCase()));
};

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(stockdata());
  }, [dispatch]);
  const show = useSelector((state) => state.stock);
  const [filterdata, setFilterdata] = useState('');
  const filterItem = handlefilter(filterdata, show.data);
  let count = 0;
  return (
    <div>
      <div className="Search">
        <input type="text" className="input-search" onChange={(e) => setFilterdata(e.target.value)} placeholder="Type company name" />
        <button type="submit" className="submit-search"><FiSearch /></button>
      </div>
      <ul className="Fetch-home">
        {filterItem.map((data) => {
          count += 1;
          return (
            <li id={data.volumn} key={count}>
              <div className="Main-data">
                <h2>{data.symbol}</h2>
                <NavLink to={`/details/:${data.symbol}/:${count}`}>
                  <button type="button" className="Button input-submit" onClick={() => dispatch(IC(data.symbol))}><FaAngleRight /></button>
                </NavLink>
              </div>
              <div className="Fetch-footer">
                <h3 className="margin">{data.companyName}</h3>
                <p className="margin">
                  $
                  {data.price}
                </p>
              </div>
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default Home;
