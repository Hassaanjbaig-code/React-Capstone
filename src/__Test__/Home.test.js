import React from 'react';
import '@testing-library/jest-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/Store';
import Home from '../Components/Home';

test('should first', async () => {
  const fetchd = await fetch('https://financialmodelingprep.com/api/v3/stock-screener?marketCapMoreThan=1000000000&betaMoreThan=1&volumeMoreThan=10000&sector=Technology&exchange=NASDAQ&dividendMoreThan=0&limit=48&apikey=41efa39e0c22f8d0149cc5c4406e677c', {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  });
  const data = await fetchd.json();
  expect(data.length).toBe(48);
});

test('should first', () => {
  const tree = renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <Home />
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
