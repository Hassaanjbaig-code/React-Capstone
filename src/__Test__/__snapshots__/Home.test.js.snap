// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<div>
  <h1>
    US Companies Stock Price
  </h1>
  <ul
    className="Fetch-home"
  />
</div>
`;
